tests/phpunit/tests/multisite/ms-functions/update_posts_count.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+if ( is_multisite() ) :
+	/**
+	 * Test update_posts_count() get called via filters of WP_Site in multisite.
+	 *
+	 * @group ms-site
+	 * @group multisite
+	 *
+	 * @covers ::_update_posts_count_on_delete
+	 */
+	class Tests_update_posts_count_on_delete extends WP_UnitTestCase {
+
+		/**
+		 * This test that the posts count is updated correctly when a posts are added and deleted.
+		 * @ticket 53443
+		 */
+		public function test_update_posts_count_on_delete() {
+
+			$blog_id = self::factory()->blog->create();
+			switch_to_blog( $blog_id );
+
+			/*
+			 * Get the total number of posts, and take into account the "Hello World" blog post automatically created with the blog.
+			 */
+			$current_post_count = (int) get_option( 'post_count' ) + 1;
```

Comment:
When [54462](https://core.trac.wordpress.org/ticket/54462) is fixed, we won't need to add 1 to this count because the Hello world! post will be accounted for

Reply:
done